feature-gate/program/src/instruction.rs

```diff
@@ -0,0 +1,87 @@
+//! Program instructions
+
+use {
+    borsh::{BorshDeserialize, BorshSchema, BorshSerialize},
+    solana_program::{
+        instruction::{AccountMeta, Instruction},
+        pubkey::Pubkey,
+        system_program,
+    },
+};
+
+/// Feature Gate program instructions
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, BorshSchema, PartialEq)]
+pub enum FeatureGateInstruction {
+    /// Submit a feature for activation.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[ws]` Feature account (must be a system account)
+    ///   1. `[s]` Authority
+    ///   3. `[]` System program
+    Activate,
+    /// Revoke a pending feature activation.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[w]` Feature account
+    ///   1. `[w]` Destination (for rent lamports)
+    ///   2. `[s]` Authority
+    Revoke,
```

Comment:
Let's make the action really clear everywhere
```suggestion
    RevokePendingActivation,
```